### PR TITLE
Enable pulseIn()

### PR DESCRIPTION
**Pull request by sorscode**
quick edit to make pulseIn() work.

### CODE REVIEW

File: cores/esp32/wiring_pulse.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+extern uint32_t xthal_get_ccount();
+
+#define WAIT_FOR_PIN_STATE(state) \
+    while (digitalRead(pin) != (state)) { \
+        if (xthal_get_ccount() - start_cycle_count > timeout_cycles) { \
+            return 0; \
+        } \
+        optimistic_yield(5000); \
```
Comment (me-no-dev) [inline cores/esp32/wiring_pulse.c]: optimistic_yield no longer needed

**Comment by me-no-dev** [comment]
did you test this?

**Comment by sorscode** [comment]
Yes, I have it running on my desk right now. I can remove the optimistic_yield.

**Comment by sorscode** [comment]
Okay, my latest commit, has removed the optimistic_yield

**Pull request merged**